nomad/job_endpoint_hooks.go

```diff
@@ -182,8 +184,11 @@ func (jobImpliedConstraints) Mutate(j *structs.Job) (*structs.Job, []error, erro
 	for _, tg := range j.TaskGroups {
 
 		// If the task group utilises Vault, run the mutator.
-		if _, ok := vaultBlocks[tg.Name]; ok {
-			mutateConstraint(constraintMatcherLeft, tg, vaultConstraint)
+		vaultTasks := maps.Keys(vaultBlocks[tg.Name])
+		sort.Strings(vaultTasks)
```

Comment:
Just out of curiosity, is the sorting relevant here?

Reply:
It makes the error messages stable, which was necessary to deflake tests.